### PR TITLE
Make the dbal event store connection configurable.

### PR DESCRIPTION
**Pull request by kimlai**
Will fix https://github.com/qandidate-labs/broadway/issues/3.

A new configuration option is available :
```
broadway:
    event_store:
        dbal:
            connection: sqlite
```
It defaults to `null`, in which case the service `database_connection` is
used (alias created by doctrine to get the default connection).

### CODE REVIEW

**Comment by cordoval** [comment]
:+1:

File: src/Broadway/Bundle/BroadwayBundle/DependencyInjection/BroadwayExtension.php [hunk 1]
```diff
@@ -107,6 +110,31 @@ private function loadEventStore(array $config, ContainerBuilder $container)
...
             $config['dbal']['use_binary']
         );
+
+        $this->dbalEventStoreConnection = $config['dbal']['connection'];
+    }
+
+    /**
+     * This needs to be done in a compiler pass, to access doctrine connections definitions.
+     */
+    public function defineDBALEventStoreConnection(ContainerBuilder $container)
```
Comment (wjzijderveld) [inline src/Broadway/Bundle/BroadwayBundle/DependencyInjection/BroadwayExtension.php]: I would prefer to move this to the actual compiler pass. You can set a parameter for the connection name.

I see you also use `$this->getAlias()`, but we can probably pass it as a constructor argument to the compiler pass or maybe even hardcode it.

File: src/Broadway/Bundle/BroadwayBundle/DependencyInjection/DefineDBALEventStoreConnectionCompilerPass.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Broadway\Bundle\BroadwayBundle\DependencyInjection;
+
+use RuntimeException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
```
Comment (wjzijderveld) [inline src/Broadway/Bundle/BroadwayBundle/DependencyInjection/DefineDBALEventStoreConnectionCompilerPass.php]: These last 2 are not used

**Comment by wjzijderveld** [comment]
Closing this one for now, #182 will use the commits from this PR with the comments resolved.
@kimlai If you still want to continue on this, please let me know!

**Comment by kimlai** [comment]
You seem to have things under control :) Sorry I wasn't able to find time to finish this.

**Comment by wjzijderveld** [comment]
No problem, just wanted to make sure I wasn't doing stuff you wanted to
finish. Thanks for the work you already put into it!

**Pull request closed without merging**